### PR TITLE
Cale/fix artifact fetches

### PR DESCRIPTION
**Pull request by cpennington**

### CODE REVIEW

File: edxpipelines/patterns/tasks.py [hunk 1]
```diff
@@ -296,6 +306,7 @@ def generate_create_ami(
     Args:
         job (gomatic.job.Job): the gomatic job which to add the launch instance task
         runif (str): one of ['passed', 'failed', 'any'] Default: passed
+        launch_info_path (str): The path to launch_info.yaml (optional)
```
Comment (rlucioni) [inline edxpipelines/patterns/tasks.py]: `launch_info_path ` is a positional argument, it isn't optional.

File: edxpipelines/patterns/jobs.py [hunk 2]
```diff
@@ -73,22 +64,19 @@ def generate_build_ami(pipeline,
         ec2_security_group_id=env_config['ec2_security_group_id'],
         ec2_instance_profile_name=env_config['ec2_instance_profile_name'],
         base_ami_id=env_config['base_ami_id'],
-        upstream_build_artifact=ami_artifact,
+        upstream_build_artifact='{}/{}'.format(
```
Comment (rlucioni) [inline edxpipelines/patterns/jobs.py]: Minor: consider naming this kwarg `ami_override_path`, for consistency with the other path argument you're adding, `launch_info_path`.

File: edxpipelines/patterns/tasks.py [hunk 3]
```diff
@@ -669,7 +669,7 @@ def fetch_edx_mktg(job, secure_dir, runif="passed"):
 
 def generate_run_app_playbook(
         pipeline, job, playbook_with_path, edp, app_repo,
-        upstream_artifact_base=None,
+        launch_artifacts_base_path=None,
```
Comment (rlucioni) [inline edxpipelines/patterns/tasks.py]: Does this name need to reference the launch task? `artifact_base_path` might be more concise.

File: edxpipelines/patterns/stages.py [hunk 4]
```diff
@@ -218,6 +223,18 @@ def generate_run_play(pipeline,
     tasks.generate_requirements_install(job, 'configuration')
     tasks.generate_target_directory(job)
 
+    for file_name in ('key.pem', 'launch_info.yml', 'ansible_inventory'):
```
Comment (rlucioni) [inline edxpipelines/patterns/stages.py]: Can you use constants here like `LAUNCH_INSTANCE_FILENAME`? I'm going to need access to these filenames, too, in https://github.com/edx/edx-gomatic/pull/201.

File: edxpipelines/patterns/tasks.py (hunk 1 above)
Comment (rlucioni) [inline edxpipelines/patterns/tasks.py]: s/launch_info.yaml/launch_info.yml

File: edxpipelines/patterns/tasks.py (hunk 3 above)
Comment (cpennington) [inline edxpipelines/patterns/tasks.py]: In this case, these are specifically launch_artifacts that it's going to retrieve (`key.pem`, `ansible_inventory`, and `launch_info.yml`).

File: edxpipelines/patterns/stages.py (hunk 4 above)
Comment (cpennington) [inline edxpipelines/patterns/stages.py]: oh, yeah, will do.

File: edxpipelines/patterns/tasks.py (hunk 3 above)
Comment (rlucioni) [inline edxpipelines/patterns/tasks.py]: Understood. Making that clear in the name makes sense.

**Pull request merged**